base/intfuncs.jl

```diff
@@ -234,6 +234,15 @@ const HWNumber = Union{HWReal, Complex{<:HWReal}, Rational{<:HWReal}}
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{2}) = x*x
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{3}) = x*x*x
 
+# don't use the inv(x) transformation here since x^p is slightly more accurate
+@inline literal_pow(::typeof(^), x::AbstractFloat, ::Val{p}) where {p} = x^p
+
+# eliminate method ambiguities:
+@inline literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{0}) = one(x)
+@inline literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{1}) = x
+@inline literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{2}) = x*x
+@inline literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{3}) = x*x*x
```

Comment:
Is it worth adding `@inline literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{-1}) = 1/x` as well?

Reply:
Why are these ambiguous? LLVM already knows about how to do this rewrite, when it can show that it won't cause accuracy issues. That at least seemed to be the conclusion in https://github.com/JuliaLang/julia/pull/19890 (otherwise, I would have also just altered the flag to let LLVM know that it was allowed to do this rewrite).

Reply:
Without these methods, `literal_pow(::typeof(^), x::AbstractFloat, ::Val{p})` is ambiguous with `literal_pow(::typeof(^), x::HWNumber, ::Val{2})` etc.

Reply:
@simonbyrne, that couldn't hurt, I guess — might as well add it.

Reply:
@vtjnash, actually, it seems like there is no longer an ambiguity — there used to be an ambiguity when I was defining `literal_pow(::typeof(^), x::Union{Float32,Float64}, ::Val{p})`, but it looks like it may be fine now that it is `AbstractFloat`.

Reply:
Hm, ok. It still seems unnecessary to be defining a special function overlay to be doing something that LLVM is doing already (e.g., it already rewrites `pow(x, 2)` as `x*x` whenever possible). The same is true of `-1` method proposed above: that optimization is already implemented transparently in the backend.

Reply:
LLVM only does that conversion for types that it knows about.  e.g. it won't do it for `Complex{Float64}` or `DecimalFloat32`.